OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -176,10 +218,19 @@ private void setCurrentUnit(CourseComponent component) {
 
     private void tryToUpdateForEndOfSequential() {
         int curIndex = pager.getCurrentItem();
+
+        if (!selectedUnit.getParent().getId().equalsIgnoreCase(pagerAdapter.getUnit(curIndex).getParent().getId())) {
+            indicatorController.initialize(getTotalComponentsCount(pagerAdapter.getUnit(curIndex)));
```

Comment:
indicatorController.**initialize**, we are not actually initializing it here, so good time to rename it to some more sanity name like `setCount()`